src/components/User.js

```diff
@@ -0,0 +1,260 @@
+        .then(result => result.data);
+}
+
+async function getListOfComments(authStr, postId) {
+    return axios.get("http://" + API_DOMAIN + "/api/comments/", { headers: { Authorization: authStr },
+        params: { post_id: postId,}})
+        .then(result => result.data);
+}
+
+function postComment(authStr, credentials) {
+    return axios.post("http://" + API_DOMAIN + "/api/comments/", credentials, {
+        headers: {Authorization: authStr}
+    }).catch((error) => {
+        console.log(error.response);
+    });
+}
+
+function postLike(authStr, postId, isLike) {
+    const postStr = ''.concat(postId);
+    if (isLike) {
+        return axios.post("http://" + API_DOMAIN + "/api/posts/" + postStr + "/like/", {}, {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+    else {
+        return axios.post("http://" + API_DOMAIN + "/api/posts/" + postStr + "/dislike/", {}, {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+}
+
+function deleteLike(authStr, postId, isLike) {
+    const postStr = ''.concat(postId);
+    if (isLike) {
+        return axios.delete("http://" + API_DOMAIN + "/api/posts/" + postStr + "/like/",  {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+    else {
+        return axios.delete("http://" + API_DOMAIN + "/api/posts/" + postStr + "/dislike/", {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+}
+
+function deletePost(authStr, postId) {
+    const postStr = ''.concat(postId);
+    return axios.delete("http://" + API_DOMAIN + "/api/posts/" + postStr + "/",  {
+        headers: {Authorization: authStr}
+    }).catch((error) => {
+        console.log(error.response);
+    });
+}
+
+function User(props) {
+    const { token, setToken } = useToken();
+    const [ postList, setPostList ] = useState([]);
+    const [ user, setUser] = useState({});
+    const [ loading, setLoading ] = useState(true);
+    const authStr = 'Token '.concat(token);
+    const { id } = useParams();
+
+    useEffect(() => {
+        const handleSome = async () => {
+            setLoading(true);
+            console.log(id);
+            console.log(''.concat(id));
+            const user = await getUserInfo(authStr, id);
+            setUser(user);
+
+            const posts = await getListOfPosts(authStr, user.id);
+            for (let post of posts) {
+                post.showComments = false;
+            }
+            setPostList({postList: posts});
+            setLoading(false);
+        }
+        handleSome();
+    }, []);
+
+
+    const clickLike = (postId, index) => {
```

Comment:
For this and other functions: let's not copy-paste them but put into a common place to follow DRY style